zerver/tests/test_message_send.py

```diff
@@ -2487,6 +2487,36 @@ def test_addressee_legacy_build_for_stream_id(self) -> None:
         stream_id = result.stream_id()
         self.assertEqual(stream.id, stream_id)
 
+    def test_addressee_legacy_build_for_bad_num_streams(self) -> None:
+        realm = get_realm("zulip")
+        self.login("iago")
+        sender = self.example_user("iago")
+
+        with self.assertRaisesRegex(JsonableError, "Missing stream"):
+            Addressee.legacy_build(
+                sender=sender,
+                message_type_name="stream",
+                message_to=[],
+                topic_name="random_topic",
+                realm=realm,
+            )
+
+        self.subscribe(sender, "Denmark")
+        stream1 = get_stream("Denmark", realm)
+        self.subscribe(sender, "Scotland")
+        stream2 = get_stream("Scotland", realm)
+        with self.assertRaisesRegex(JsonableError, "Cannot send to multiple streams"):
+            Addressee.legacy_build(
+                sender=sender,
+                message_type_name="stream",
+                message_to=[stream1.id, stream2.id],
+                topic_name="random_topic",
+                realm=realm,
+            )
```

Comment:
In general, we should be testing the `send_message` API, not this `legacy_build` code path itself, which we're hoping to clean up down the line. 

I don't think that strategy is possible for `Cannot send to multiple streams`, because that case is likely impossible. We should probably replace this test of an unreachable `if / JsonableError` construction with just having an `assert` statement in the runtime code. It's possible the same applies to the other cases in this test.